src/moaicore/MOAICCParticle.cpp

```diff
@@ -0,0 +1,218 @@
+//
+//  MOAICCParticle.cpp
+//  libmoai
+//
+//  Created by Isaac Barrett on 3/4/15.
+//
+//
+
+#include <stdio.h>
+#include <moaicore/MOAICCParticle.h>
+#include <moaicore/MOAICCParticleSystem.h> 
+
+// default constructor
+MOAICCParticle::MOAICCParticle()
+    :mTimeToLive(0.0f),
+    mDeckIndex(0),
+    mCurrentPosition(0.0f, 0.0f),
+    mStartPosition(0.0f, 0.0f),
+    mParticleSize(0.0f),
+    mDeltaParticleSize(0.0f),
+    mParticleRotation(0.0f),
+    mDeltaParticleRotation(0.0f),
+    mDirection(0.0f, 0.0f),
+    mGravity(0.0f, 0.0f),
+    mRadialAcceleration(0.0f),
+    mTangentialAcceleration(0.0f),
+    mAngle(0.0f),
+    mDegreesPerSecond(0.0f),
+    mRadius(0.0f),
+    mDeltaRadius(0.0f),
+    mRotationalAcceleration(0.0f)
+{
+    for (int index = 0; index < 4; ++index) {
+        this->mColor[index] = 0.0f;
+        this->mDeltaColor[index] = 0.0f;
+    }
+}
+
+// initialization constructor from a MOAICCParticleSystem
+MOAICCParticle::MOAICCParticle(MOAICCParticleSystem *particleSystem) {
+    // timeToLive
+    this->mTimeToLive = USFloat::Rand(particleSystem->mMinLifespan, particleSystem->mMaxLifespan);
+ 
+    // position
+    this->mStartPosition.mX = particleSystem->mSourcePos[0] + particleSystem->mSourcePosVariance[0] * USFloat::Rand(-1.0f, 1.0f);
+    this->mStartPosition.mY = particleSystem->mSourcePos[1] + particleSystem->mSourcePosVariance[1] * USFloat::Rand(-1.0f, 1.0f);
+    
+    // color
+    float start;
+    float end;
+    
+    for (int i = 0; i < 4; ++i) {
+        start = USFloat::Clamp(particleSystem->mStartColor[i] + particleSystem->mStartColorVariance[i] * USFloat::Rand(-1.0f, 1.0f), 0.0f, 1.0f);
+        
+        end = USFloat::Clamp(particleSystem->mFinishColor[i] + particleSystem->mFinishColorVariance[i] * USFloat::Rand(-1.0f, 1.0f), 0.0f, 1.0f);
+        
+        this->mColor[i] = start;
+        
+        this->mDeltaColor[i] = (end - start) / this->mTimeToLive;
+    }
+    
+    // partice size
```

Comment:
![fix-it-felix](https://cloud.githubusercontent.com/assets/245857/6679890/2eec7ccc-cc0c-11e4-871c-be9ef983245a.jpeg)